### DISCUSSION TITLE
Have to load /upgrade.php multiple times when upgrading website

### DISCUSSION THREAD

**Discussion by roughnecks** [xenocrat/chyrp-lite#260] [Q&A]
Hello,
I have a personal blog with chyrp-lite and then another blog for our xmpp-it community.

When I do upgrade both sites, mine works fine after I open /upgrade.php and run it, while the other site needs me to run  the upgrade page multiple times before everything works again.

At some point I changed from chyrp-lite release to master on git for that site (because we needed a fix only available there), but then reverted again to stable releases: could this be an issue?
When I get errors after the initial upgrade, trying to navigate the site, even hard-reloading that page doesn't work, so I usually hit the upgrade button once again, till it finally works.

This was a bit scary the first time around, since I thought to have screwed up.. We have backups though :)

Also thanks for your work, chyrp-lite rocks!

**Comment by xenocrat** [comment]
Hello there,

The upgrader is version-agnostic and designed to be idempotent so it can be run safely multiple times, but it shouldn't be necessary to do so. The upgrader runs in "debug mode" with strict error reporting, meaning that almost any error at all will stop it. If you're not seeing an error page then the upgrader has not encountered any known issues. I try to keep the master branch in good order; inevitably there are times when a horrible mistake goes unnoticed, but that shouldn't cause any irreversible problems.

A few questions:
- You wrote "before everything works again" - what errors are you seeing?
- Any errors or warnings reported in *error_log.txt*?
- Any differences in your upgrade process for the two sites?

Without knowing more, my first thoughts are regarding cache files. A stale Twig cache could be causing some issues if not removed; if upgrading using the upgrade instructions this won't be an issue, but if you upgrade by doing an overwrite in place, the Twig cache won't be purged. It is located at */includes/caches/twig/*. Purge it by deleting all the sub-directories within it.

Glad to read that you backup before upgrading. The warning is there for a reason! :-D

**Comment by roughnecks** [comment]
1. I believe Bad Gateway errors, I didn't think of reporting this earlier since I managed to get it working 2 times with the same trick.. and I believe it's also written somewhere that the upgrade script might be run more than once.
2. error_log.txt contains just a line repeated a few times: `[20-May-2024 12:21:44 UTC] LOCALE it_IT.UTF-8`
3. No differences between the upgrade process and both are hosted on the same VPS

OK, I removed the twig cache after it worked eventually (I knew about this since you told me it in another issue about themes modding)
Next time I'll delete all of the cache before running the upgrade and we'll see.

👍

**Comment by xenocrat** [comment]
Chyrp Lite doesn't currently issue 502 errors from the application level for any reason, so the 502 must be coming from lower down server stack. I found [this article](https://gridpane.com/kb/diagnosing-502-errors/) suggesting a few common causes and troubleshooting options. It could be that the upgrade tasks cause excessive database locking, which then takes a little while to reset. However I would expect to see the upgrade process itself have trouble in that case.

Let me know if it happens again and you are able to gather any insights on the probable cause.

**Comment by roughnecks** [comment]
Will do, thanks! I'm closing this for the time being.
